flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/DataStream.scala

```diff
@@ -750,6 +750,34 @@ class DataStream[T](stream: JavaStream[T]) {
   }
 
   /**
+    * Assigns timestamps to the elements in the data stream and periodically creates
+    * watermarks to signal event time progress.
+    *
+    * This method is used for cases where the programmer wants the Watermarks emitted in
+    * the stream to lag behind the maximum timestamp seen so far by a fixed period of time.
+    *
+    * For more general cases, use the general methods
+    * [[assignTimestampsAndWatermarks(AssignerWithPeriodicWatermarks)]]
+    * and [[assignTimestampsAndWatermarks(AssignerWithPunctuatedWatermarks)]].
+    *
+    * @param fixedWatermarkTrailingPeriod the fixed lag between the currently emitted
+    *                                     watermark and the maximum timestamp seen so far.
+    * @param extractor the method to extracts the timestamp from a given record.
+    */
+  @PublicEvolving
+  def assignTimestampsWithAFixedPeriodTrailingWatermark(fixedWatermarkTrailingPeriod: Long,
```

Comment:
It is true, although I added it because there was already a method for the ascending one. If there is a consensus, I can remove it.